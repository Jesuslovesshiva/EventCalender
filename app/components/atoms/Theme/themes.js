/* eslint-disable sort-keys */
const baseColors = {
  white: '#FFFFFF',
  silver: '#F3F5F6',
  gray: {
    light: '#E5E7E7',
    medium: '#BBBBBB',
    dark: '#212121'
  },
  red: {
    light: '#F8CED9',
    medium: '#DB0A40'
  },
  orange: {
    light: '#FCEACE',
    medium: '#F2940A'
  },
  yellow: {
    light: '#FCF6D0',
    medium: '#F2D014'
  },
  green: {
    light: '#E5F0DF',
    medium: '#7CB45E'
  },
  emerald: {
    light: '#D6E7DF',
    medium: '#32885F'
  },
  blue: {
    light: '#CEE7F5',
    medium: '#0a86cb',
    dark: '#0a3064'
  }
}

const standard = {
  colors: {
    ...baseColors,
    accent: baseColors.red.medium,
    backdrop: '#081D39',
    border: baseColors.gray.dark,
    primary: baseColors.blue.dark,
    background: baseColors.silver,
    divider: baseColors.gray.medium,
    surface: baseColors.white
  }
}

export {
  // eslint-disable-next-line import/prefer-default-export
  standard
}
