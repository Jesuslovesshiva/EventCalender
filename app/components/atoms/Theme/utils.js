const flattenObject = (sourceObject, separator = '-', keyPrefix = null) =>
  Object.entries(sourceObject).reduce((acc, [key, value]) => {
    const newKey = [keyPrefix, key].filter(Boolean).join(separator)

    if (typeof value === 'object') {
      return {
        ...acc,
        ...flattenObject(value, separator, newKey)
      }
    }

    return {
      ...acc,
      [newKey]: value
    }
  }, {})

const hexToRgb = hexColor =>
  hexColor
    .replace('#', '')
    .match(/.{1,2}/gu)
    .map(hexColorValue => parseInt(hexColorValue, 16))
    .join(' ')

export {flattenObject, hexToRgb}
