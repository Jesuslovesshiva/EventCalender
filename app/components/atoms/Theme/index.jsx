import mapValues from 'lodash/mapValues'
import {useMemo} from 'react'

import * as themes from './themes'
import {flattenObject, hexToRgb} from './utils'

const Theme = () => {
  const css = useMemo(() => {
    const cssVariables = {
      ...themes.standard,
      colors: mapValues(flattenObject(themes.standard.colors, '-'), hexToRgb)
    }

    return Object.entries(flattenObject(cssVariables, '-'))
      .map(([key, value]) => `--${key}: ${value};`)
      .join('\n')
  }, [])

  return (
    <style>
      {`:root {
        ${css}
      }`}
    </style>
  )
}

export default Theme
