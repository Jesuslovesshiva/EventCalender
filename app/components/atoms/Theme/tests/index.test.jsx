import render from '@/utils/test-utils/custom-render'

import Theme from '../index'

const renderComponent = () => render(<Theme />)

describe('Theme', () => {
  it('should match the snapshot', () => {
    const {container} = renderComponent()
    expect(container.firstChild).toMatchSnapshot()
  })
})
