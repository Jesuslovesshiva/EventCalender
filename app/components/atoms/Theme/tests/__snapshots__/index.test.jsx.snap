// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme should match the snapshot 1`] = `
<style>
  :root {
        --colors-white: 255 255 255;
--colors-silver: 243 245 246;
--colors-gray-light: 229 231 231;
--colors-gray-medium: 187 187 187;
--colors-gray-dark: 33 33 33;
--colors-red-light: 248 206 217;
--colors-red-medium: 219 10 64;
--colors-orange-light: 252 234 206;
--colors-orange-medium: 242 148 10;
--colors-yellow-light: 252 246 208;
--colors-yellow-medium: 242 208 20;
--colors-green-light: 229 240 223;
--colors-green-medium: 124 180 94;
--colors-emerald-light: 214 231 223;
--colors-emerald-medium: 50 136 95;
--colors-blue-light: 206 231 245;
--colors-blue-medium: 10 134 203;
--colors-blue-dark: 10 48 100;
--colors-accent: 219 10 64;
--colors-backdrop: 8 29 57;
--colors-border: 33 33 33;
--colors-primary: 10 48 100;
--colors-background: 243 245 246;
--colors-divider: 187 187 187;
--colors-surface: 255 255 255;
      }
</style>
`;
